script/build-script/post-build-steps.sh

```diff
@@ -0,0 +1,161 @@
+
+# check if we are in the software-arm-incubation repo and search there
+sbLibVersionHeaderDir="${scriptDir}"/"${scriptPathDepth}"/"${sbLibSubModuleRepoFolder}"/"${sbLibVersionSubDir}"
+sbLibVersionWithDot=$(get_sbLibVersionWithDot "$sbLibVersionHeaderDir")
+
+if [ -z "${sbLibVersionWithDot}" ]; then
+  # check if we are in the sblib repo and search there
+  sbLibVersionHeaderDir="${scriptDir}"/"${scriptPathDepth}"/"${sbLibVersionSubDir}"
+  sbLibVersionWithDot=$(get_sbLibVersionWithDot "$sbLibVersionHeaderDir")
+
+  # check if sbLibVersionWithDot was not found
+  if [ -z "${sbLibVersionWithDot}" ]; then
+    sbLibVersionWithDot="x.x"
+    echo "Selfbus library version was not found in the repository, set to "${sbLibVersionWithDot}""
+  fi
+fi
+
+echo "Creating .hex/.bin files and adding sblib version "${sbLibVersionWithDot}" to filename"
+
+outputFileName="${buildArtifactFileBaseName}"
+# append build config name and app version
+outputFileName="${outputFileName}"_"${appConfigName}"_"${appVersionPrefix}""${appVersion}"
+# replace spaces with underscores
+outputFileName="${outputFileName// /_}"
+# replace missing options (--)
+outputFileName="${outputFileName//--/_}"
+# fix trailing missing option
+outputFileName="${outputFileName//-_/_}"
+# fix leading missing option
+outputFileName="${outputFileName//_-/_}"
+# replace duplicate underscores
+outputFileName="${outputFileName//__/_}"
+# convert to lowercase
+outputFileName=`echo "${outputFileName}" | tr '[:upper:]' '[:lower:]'`
+# add lib version
+outputFileName="${outputFileName}_${sbLibprefix}${sbLibVersionWithDot}"
+echo "${outputFileName}"
+
+# post-build-step from eclipse
+arm-none-eabi-size "${buildArtifactFileName}"
+# create .bin file
+arm-none-eabi-objcopy -v -O binary "${buildArtifactFileName}" "${outputFileName}.bin"
+# create .hex file
+arm-none-eabi-objcopy -v -O ihex "${buildArtifactFileName}" "${outputFileName}.hex"
+
+# check if hexDir is defined
+if [ -z "${hexDir}" ]; then
+    echo "Workspace build variable hexDir not defined or empty, exiting post-build-script."
+    exit 1
+fi
+
+# add release/debug to hex output directory
+case ${outputFileName} in
+  *"${debug}"*)
+    hexDir="${hexDir}"/"${debug}"
+    ;;
+
+  *"${release}"*)
+    hexDir="${hexDir}"/"${release}"
+    ;;
+esac
+
+# check adding flashstart to hex output directory
+if [[ "${outputFileName}" =~ .*"${flashstart}".* ]]; then
+  hexDir="${hexDir}"/"${flashstart}"
+fi
+
+# check for .hex output directory
+if [ ! -d "${hexDir}" ]; then
+  echo "Creating "${hexDir}""
+  mkdir -p "${hexDir}"
+fi
+
+cp --verbose "${currentWorkingDirectory}"/"${outputFileName}.hex" "${hexDir}"/
+
+# Do not activate checksum, not sure why, but at least .hex files get corrupted
+# see also https://community.nxp.com/t5/Blogs/Hex-file-settings-in-MCUxpresso/bc-p/1131124/highlight/true#M53
+# add checksums
+#checksum -p ${targetChip} -d "${outputFileName}.bin"
+#checksum -p ${targetChip} -d "${outputFileName}.hex"
```

Comment:
Leaving the general feedback here on the last line.

Thanks for the idea and the implementation of this output file naming approach! I like it and am happy to get it in, in general. Nevertheless, I'm afraid this approach can easily cause some grief:

1. App version needs to be set in MCUXpresso macros and also in code. It will be easy to miss one of them, and the result file names will not match the file's content anymore. ⚡ 
2. SBLib version is read from the git submodule's directory. This means
   * the submodule must have been retrieved and be up-to-date ⚡ , and
   * must align with the code that was actually linked into the binary ⚡ . Remember that right now we actually have SBLib cloned next to the software-arm-incubation repo, so we're always linking the "work" repository binaries whereas we put the "submodule" repository version number in the file name.

Do we have any ideas about how to address these issues, or even already plans to do so?

Reply:
I totally agree with you on all your points. I see this script only as a workaround (or step 0.0.1) to an automated build system. Right now we have nothing. The lib/app version is not mentioned in the filename, only the lib version is hardcoded inside the firmware, which is also prone to be invalid like you mentioned in your first point.

The biggest advantage I see in the script is that, if configured correctly, all `*.hex` files end up sorted in one output folder instead of having to copy them manually from the projects subfolders.

From another project I got the idea of using the last digits of the commit-sha of the lib and the app in addition to a hard-coded version, but that won't eliminate your second point.

Reply:
Alright, thanks for your follow-up.